src/workspace/Lesson/hooks.js

```diff
@@ -1,282 +1,91 @@
+import { fetchLessonWithSections, lessonWasUpdated, saveLesson, setLesson } from "~/Contexts/lesson.actions"
+import { unsavedChanges , newError } from '~/Contexts/notifications.actions'
+import { mergeSerloDiff, saveSections, sectionWasUpdated, setSections, fetchSection } from '~/Contexts/section.actions'
+import { serverApi } from "~/utils/api"
+import { loadLessonData, loadSectionData } from "~/utils/cache"
 import { editorWS } from "~/utils/socket"
-import { loadEditorData, saveEditorData } from "~/utils/cache"
-import { buildDiff } from "~/utils/diff"
-import { createSection } from "~/actions/section"
 
 export function useBootstrap(id, courseId, dispatch, dispatchUserAction) {
-    async function fetchData() {
+    async function bootstrap() {
         try {
             const user = await serverApi.get("/me")
             dispatchUserAction({ type: "BOOTSTRAP_USER", payload: user })
         } catch (err) {
             console.warn("Could not fetch user data")
+            dispatch(newError('Es konnten nicht alle Daten geladen werden, eventuell kommt es zu einschränkungen'))
         }
 
-        try {
-            const cacheData = loadEditorData(id)
-            let lesson
-            if (
-                cacheData &&
-                cacheData.hasOwnProperty("lesson") &&
-                (cacheData.savedToBackend === false || editorWS.connected)
-            ) {
-                lesson = cacheData.lesson
-            } else {
-                lesson = await editorWS.emit(
-                    'get',
-                    `course/${courseId}/lessons`,
-                    id,
-                    {all: 'true'}
-                )
-
-                lesson.id = lesson._id
-                if(lesson.sections.length === 0){
-                    const section = await editorWS.emit('create', `lesson/${id}/sections`, {})
-                    lesson.sections.push(section)
-                }
-                lesson.sections = lesson.sections.map(section => ({
-                    ...section,
-                    id: section._id,
-                    docValue: section.state,
-                    savedDocValue: section.state,
-                    notes: section.note,
-                    visible: true, // TODO: remove should be set by server and blur mode should removed
-                }))
+        const cachedLessonData = loadLessonData(id)
+		if (
+			cachedLessonData &&
+			Object.prototype.hasOwnProperty.call(cachedLessonData, 'sections') &&
+			(cachedLessonData.savedToBackend === false || !editorWS.isConnected)
+		) {
+            // TODO: compare timestamps and hash with server state and save if possible or set
+            // saved to true if hash is the same (needs server route, to do not send all data)
+            console.log('Data are loaded from cach', !cachedLessonData.savedToBackend)
+            dispatch(setLesson(cachedLessonData))
+            const sections = loadSectionData(...cachedLessonData.sections)
```

Comment:
Mhh sollte man evtl. in loadCache ...umbenennen, oder cache.loadSectionData lassen. Damit klar ist das es aus dem cache kommt.